packages/replay/src/main.ts

```diff
@@ -458,16 +460,28 @@ async function processRecording(id: string, opts: Options = {}) {
   return succeeded ? recordingId : null;
 }
 
-async function uploadAllRecordings(opts: Options & FilterOptions = {}) {
+async function uploadAllRecordings(opts: Options & UploadOptions = {}) {
   const server = getServer(opts);
   const dir = getDirectory(opts);
   const recordings = filterRecordings(readRecordings(dir), opts.filter);
 
-  const recordingIds = await Promise.allSettled(
-    recordings
-      .filter(r => !uploadSkipReason(r))
-      .map(r => doUploadRecording(dir, server, r, opts.verbose, opts.apiKey, opts.agent))
-  );
+  if (opts.batchSize) {
+    debug("Batching upload in groups of %d", opts.batchSize);
+  }
+
+  const batchSize = Math.min(opts.batchSize || 20, 25);
+
+  let recordingsToUpload = recordings.filter(r => !uploadSkipReason(r));
+
+  let recordingIds: (string | null)[] = [];
+  while (recordingsToUpload.length > 0) {
+    const batch = recordingsToUpload.splice(0, batchSize);
+    const batchIds = await Promise.allSettled(
```

Comment:
The main downside of this implementation is that _all_ handlers in a batch need to finish before the next batch will upload. Might be worth using something like https://github.com/sindresorhus/p-map since it has a `concurrency` option.

Reply:
Works for me!

Reply:
That was a nice addition. Thanks, @loganfsmyth ! Added in 06f3ded